src/Microsoft.Security.Utilities.Core/PreciselyClassifiedSecurityKeys/LegacyCommonAnnotatedSecurityAccessKey.cs

```diff
@@ -10,12 +10,13 @@ public abstract class LegacyCommonAnnotatedSecurityAccessKey : UnclassifiedLegac
     {
         abstract protected string ProviderSignature { get; }
 
-        protected virtual string PlatformData => $"[{WellKnownRegexPatterns.Base62}]{{12}}";
+        private const string PlatformData = $"[{WellKnownRegexPatterns.Base62}]{{12}}";
 
-        protected virtual string ProviderData => $"AAAA";
+        protected string ProviderData { get; }
 
-        public LegacyCommonAnnotatedSecurityAccessKey()
+        protected LegacyCommonAnnotatedSecurityAccessKey(string providerData = "AAAA")
         {
+            ProviderData = providerData;
```

Comment:
The bug on the work in progress was around here. Base constructor was using ProviderData *before* it was overwritten by derived classes.